samples: openthread: Adding HCI capabilities in RCP sample. (#12904)
nrfconnect/sdk-nrf: samples/openthread/coprocessor/README.rst

.. _ot_coprocessor_testing:

> HCI support

Reviewer: Could you please extend this a bit explaining the use case for this feature? Also we might want to extend Thread architectures overview page: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/thread/overview/architectures.html#radio-co-processor-rcp

Author: Jira ticket KRKNWK-18076

===========

Currently HCI is supported only via nRF native USB interface. The device will show two virtual UART ports.
Usually the first one will be associated with HCI interface, the second one with Thread coprocessor.

Testing
=======

After building the sample and programming it to your development kit, complete the following steps to test it:

1. Connect the development kit's SEGGER J-Link USB port or native USB port to the PC USB port with a USB cable.
#. Get the kit's serial port name (for example, :file:`/dev/ttyACM0`).
#. Run and configure ot-cli as described in :ref:`ug_thread_tools_ot_apps`.
#. From this point, you can follow the :ref:`ot_cli_sample_testing` instructions in the CLI sample by removing the `ot` prefix for each command.

> #. If you are using HCI please follow :ref`zephyrt:samples:bluetooth:hci_uart:README`:

Reviewer 2: Is this meant to be step that happens after following the instructions for the CLI sample?

```suggestion
#. If you are using HCI, follow the instructions for the :ref`zephyrt:bluetooth-hci-uart-sample` sample in the Zephyr documentation.
```
Which section of the sample documentation is the user meant to follow?

.. _ot_coprocessor_testing:

HCI support
===========

Currently, HCI is only supported using the nRF USB interface.
The device will show two virtual UART ports.
Usually the first port will be associated with the HCI interface, and the second one with the Thread co-processor.

Testing
=======

After building the sample and programming it to your development kit, complete the following steps to test it:

1. Connect the development kit's SEGGER J-Link USB port to the PC USB port with a USB cable.
   If you are using HCI, connect the kit's nRF USB port to the PC USB port instead.
#. Get the kit's serial port name (for example, :file:`/dev/ttyACM0`).
#. Run and configure ot-cli as described in :ref:`ug_thread_tools_ot_apps`.
#. From this point, you can follow the :ref:`ot_cli_sample_testing` instructions in the CLI sample by removing the `ot` prefix for each command.

> #. If you are using HCI, follow the instructions for the :ref`zephyrt:bluetooth-hci-uart-sample` sample in the Zephyr documentation.

Reviewer 2: ```suggestion
#. If you are using HCI, follow the instructions for the :ref:`zephyrt:bluetooth-hci-uart-sample` sample in the Zephyr documentation.
```
Is there a specific section the user should read in the Zephyr sample documentation, and is this step intended to be done after or instead of the CLI instructions.

Author: Last two steps can be done in any order. work of HCI and RCP should be more or less independent.

Reviewer 2: Is there an option to only do the Zephyr instructions when working with HCI?

Author: Yes there is.

> #. From this point, you can follow the :ref:`ot_cli_sample_testing` instructions in the CLI sample by removing the `ot` prefix for each command.
> #. If you are using HCI, follow the instructions for the :ref:`zephyrt:bluetooth-hci-uart-sample` sample in the Zephyr documentation.

Reviewer 2: ```suggestion
#. From this point, you can follow the :ref:`ot_cli_sample_testing` instructions in the CLI sample by removing the `ot` prefix for each command.

   If you are using HCI, follow the instructions for the :ref:`zephyrt:bluetooth-hci-uart-sample` sample in the Zephyr documentation. You can follow these instead of or in addition to the CLI sample instructions.
```

Reviewer 2: ```suggestion
#. From this point, you can follow the :ref:`ot_cli_sample_testing` instructions in the CLI sample by removing the `ot` prefix for each command.

   If you are using HCI, follow the instructions for the :ref:`zephyrt:bluetooth-hci-uart-sample` sample in the Zephyr documentation.
   You can follow these instead of or in addition to the CLI sample instructions.
```
Correcting an error in my own suggestion.

>    If you are using HCI, follow the instructions for the :ref:`zephyr:bluetooth-hci-uart-sample` sample in the Zephyr documentation. You can follow these instead of or in addition to the CLI sample instructions.

Reviewer 2: ```suggestion
   If you are using HCI, follow the instructions for the :ref:`zephyr:bluetooth-hci-uart-sample` sample in the Zephyr documentation.
   You can follow these instead of or in addition to the CLI sample instructions.
```
My mistake on this one.
